.github/workflows/upload.yml

```diff
@@ -0,0 +1,49 @@
+name: Upload Artifacts
+
+on:
+  workflow_run:
+    workflows:
+      - CI
+    types:
+      - completed
+    branches:
+      - main
+
+jobs:
+  upload:
+    if: ${{ github.repository_owner == 'oven-sh' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download Artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: bun
+          pattern: bun-*
+          merge-multiple: true
+          github-token: ${{ github.token }}
+          run-id: ${{ github.event.workflow_run.id }}
+      - name: Upload to GitHub Releases
+        uses: softprops/action-gh-release@v2
+        with:
+          tag_name: canary
+          make_latest: false
+          generate_release_notes: false
+          files: bun/*
+      - name: Upload to S3 (using SHA)
+        uses: shallwefootball/s3-upload-action@4350529f410221787ccf424e50133cbc1b52704e
```

Comment:
I don't love relying on this random github action that hasn't been updated in a year https://github.com/shallwefootball/upload-s3-action